File: 프로그래머스/0/120905. n의 배수 고르기/n의 배수 고르기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int n, int numlist[], size_t numlist_len) {
 
    int* answer = (int*)malloc(numlist_len);
    int len = 0;
    
    for(int i=0;i<numlist_len;i++){
        if(numlist[i] % n == 0){
            answer[len++] = numlist[i];
        } 
    }
    answer[len] = NULL;
    return answer;
}

File: 프로그래머스/0/120905. n의 배수 고르기/n의 배수 고르기.java
import java.util.Arrays;

class Solution {
    
    public static int[] removeElement(int[] numlist,int item){
        return Arrays.stream(numlist)
            .filter(i -> i != item)
            .toArray();
    }
    
    public int[] solution(int n, int[] numlist) {
        int[] answer = new int[numlist.length];
        
        for(int i=0;i<numlist.length;i++){
            answer[i] += numlist[i] % n == 0 ? numlist[i] : 0;
        }

        int item = 0;
        answer = removeElement(answer,item);
        
        return answer;
    }
}